### ISSUE TITLE
Make Python 3.9 a requirement

### ISSUE THREAD

**Issue by enryH** [KonradUdoHannes/data-steps#4]
I found out now that the package uses features from Python 3.9, namely dict merge `{} |{}`

Example error

```python
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-15-931bfb491488> in <module>
      6 
      7 @data.step
----> 8 def apply_log(df,log_fct=np.log2):
      9     """Apply log-function elementwise to DataFrame."""
     10     return log_fct(df)

~\Anaconda3\envs\env\lib\site-packages\data_steps\single_frame.py in step(self, function, priority, active, **kwargs)
    143             return register_function
    144 
--> 145         return register_function(function)
    146 
    147     @property

~\Anaconda3\envs\env\lib\site-packages\data_steps\single_frame.py in register_function(func)
    137 
    138         def register_function(func):
--> 139             self._steps.update_step(func, priority=priority, active=active, **kwargs)
    140             return func
    141 

~\Anaconda3\envs\env\lib\site-packages\data_steps\single_frame.py in update_step(self, func, priority, **kwargs)
     62         active = kwargs.pop("active", True)
     63         if active:
---> 64             self._add_step(func, priority, **kwargs)
     65         elif func.__name__ in self._collection:
     66             self._remove_step(func)

~\Anaconda3\envs\env\lib\site-packages\data_steps\single_frame.py in _add_step(self, func, priority, **kwargs)
     67 
     68     def _add_step(self, func, priority, **kwargs):
---> 69         self._collection[func.__name__] = Step(priority, func, **kwargs)
     70 
     71     def _remove_step(self, func):

~\Anaconda3\envs\env\lib\site-packages\data_steps\single_frame.py in __init__(self, priority, function, has_secondary_result)

~\Anaconda3\envs\env\lib\site-packages\data_steps\single_frame.py in __post_init__(self)
     25         combined_arg_defaults = reversed(list(zip(reversed(argspec.args), reversed(args_defaults))))
     26 
---> 27         self.function_kwargs = {
     28             arg: value for arg, value in combined_arg_defaults
     29         } | kwonly_defaults

TypeError: unsupported operand type(s) for |: 'dict' and 'dict'
```

I think I won't update my current project now from python 3.8 to python 3.9 in fear of possible side-effects...

**Comment by KonradUdoHannes** [comment]
Thanks for pointing that out. At this point I'll mark it as a bug as the `setup.cfg` states python 3.8 as a requirements. I'll align the CI with that.

**Comment by KonradUdoHannes** [comment]
Fixed with 0.3.2. Codebase is now consistent with the pthon 3.8 requirement in the setup.cfg. Also the CI pipeline uses 3.8.10 for tests now to catch accidental use of unsupported features.
